src/include/rimage/sof/user/manifest.h

```diff
@@ -225,6 +225,27 @@ struct sof_man_module_manifest {
 	uint32_t text_size;
 };
 
+/*
+ *  Api version for sof loadable modules
+ */
+typedef union sof_module_api_version
+{
+    uint32_t full;
+    struct
+    {
+        uint32_t minor    : 10;
+        uint32_t middle   : 10;
+        uint32_t major    : 10;
+        uint32_t reserved : 2;
+    } fields;
+} sof_module_api_version;
+
+typedef const struct
+{
+    uint32_t FORMAT;
+    sof_module_api_version API_VERSION_NUMBER;
+} sof_module_build_info;
+
```

Comment:
Any reason for typedefs and member names in capitals ? Btw, we will need to pack these.

Reply:
This structure is used in iadk binaries also. Will packing impact closed source?

Reply:
ok, if it needs to go in main FW tree pls close here. It will need to be packed so all compilers treat it the same (i.e. there is no ambiguous sizing).